autorest/validations/validations.go

```diff
@@ -0,0 +1,415 @@
+/*
+Package validations provides methods for validating parameter value using reflection.
+*/
+package validations
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+)
+
+// Constraint stores constraint name, target field name
+// Rule and chain validations.
+type Constraint struct {
+
+	// Target field name for validation.
+	Target string
+
+	// Constraint name e.g. Minlength, Maxlength, Pattern, etc.
+	Name int
+
+	// Rule for constraint e.g. greater than 10, less than 5 etc.
+	Rule interface{}
+
+	// Chain Validations for struct type
+	Chain []Constraint
+}
+
+// Validation stores parameter-wise validation.
+type Validation struct {
+	TargetValue interface{}
+	Constraints []Constraint
+}
+
+// Constraints
+const (
+	InclusiveMaximum = iota + 1
```

Comment:
These constraints are defined as enum in code generator tool (C#). 
```C#
public enum Constraint
    {
        None,  // 0
        InclusiveMaximum, // 1
        ExclusiveMaximum, // 2 and so on
        InclusiveMinimum,
        ExclusiveMinimum,
        MaxLength,
        MinLength,
        Pattern,
        MaxItems,
        MinItems,
        UniqueItems,
        MultipleOf
    }
```

so I kept the same order here when defining with iota. 

```Go
const (
 +	InclusiveMaximum = iota + 1 // 1
 + ExclusiveMaximum  // 2 and so on
 +	InclusiveMinimum
 +	ExclusiveMinimum
 +	MaxLength
 +	MinLength
 +	Pattern
 +	MaxItems
 +	MinItems
 +	UniqueItems
 +	MultipleOf
 +	Null
 +	Empty
 +	ReadOnly
 +)
```